File: app/src/main/java/com/example/myapplication/Daos/TemplateDao.java
package com.example.myapplication.Daos;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.myapplication.Models.Template;

@Dao
public interface TemplateDao {
    @Insert
    void insert(Template template);

    @Update
    void update(Template template);

    @Delete
    void delete(Template template);

    @Query("DELETE FROM templates")
    void deleteAllTemplates();
}

File: app/src/main/java/com/example/myapplication/Daos/TemplateDao.kt
package com.example.myapplication.Daos

import androidx.lifecycle.LiveData
import androidx.room.*
import com.example.myapplication.Models.Template

@Dao
interface TemplateDao {
    @Insert
    fun insert(template: Template?)

    @Update
    fun update(template: Template?)

    @Delete
    fun delete(template: Template?)

    @Query("DELETE FROM templates")
    fun deleteAllTemplates()

    @Query("SELECT * FROM templates WHERE id = :id")
    fun getTemplateById(id: Int): Template?

    @get:Query("SELECT * FROM templates ORDER BY id ASC")
    val allTemplates: LiveData<List<Template?>?>?
}